pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -801,21 +643,32 @@ func handleModify(ctx *verifierContext, config *types.VerifyImageConfig,
 		config.Name, status.RefCount, config.RefCount)
 }
 
+// handleDelete means volumemgr wants us to delete the file.
+// Note that verifier will retain the file even if RefCount in VerifyImageConfig
+// is set to zero.
 func handleDelete(ctx *verifierContext, status *types.VerifyImageStatus) {
 
 	log.Infof("handleDelete(%s) objType %s refcount %d",
-		status.ImageID, status.ObjType, status.RefCount)
+		status.ImageSha256, status.ObjType, status.RefCount)
 
 	if status.ObjType == "" {
 		log.Fatalf("handleDelete: No ObjType for %s",
-			status.ImageID)
+			status.ImageSha256)
+	}
+	_, err := os.Stat(status.FileLocation)
+	if err == nil {
+		log.Infof("handleDelete removing %s",
+			status.FileLocation)
+		if err := os.RemoveAll(status.FileLocation); err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		log.Errorf("handleDelete: Unable to delete %s:  %s",
+			status.FileLocation, err)
```

Comment:
I think this should be a warning rather than an error. Calling delete is not about deleting a file, as ensuring a file does not exist. If it wasn't there in the first place, then we are at the good, desired state. Warning should be enough for that.

Reply:
Fixed